### DISCUSSION TITLE
Autostart

### DISCUSSION THREAD

**Discussion by DiagnozHub** [abhinavsingh/proxy.py#1297] [General]
Hey )

How can I start proxy.py when Linux starting ?

I tried crontab, like: 
@reboot sleep 60 && /usr/lib/meteo/carsdetector.sh

in carsdetector.sh:
proxy --hostname 0.0.0.0 --port 8766 --basic-auth usr1:pass1 --log-file /usr/lib/meteo/2.log

When I start this manually it works, but from cron it doesn't works ((

**Comment by abhinavsingh** [comment]
1. You should be able to make it work via cron too, though I haven't tried this myself ever
2. In production environment, here are 2 ways I have used:
   - Use `proxy.py` docker container.  You can easily setup certain containers to always start on boot
   - Use a process manager like `supervisord`, `system.d` to manage `proxy.py` lifecycle.  These process managers will automatically start the configured process for you.

Unfortunately as of today there is no native integration with any process manager (as that would be an external dependency).

Let me know if this doesn't help you kick start.

**Reply by DiagnozHub** [reply]
I resolved:
# Proxy server for access inner resources

cd /usr/local/lib/python3.8/dist-packages
python3.8 -m proxy --hostname 0.0.0.0 --port 8766 --basic-auth usr1:pass1 --log-file /var/log/proxy.log
